### PR TITLE
NEXUS-17816: html encode exception strings

### PR DESCRIPTION
**Pull request by dbradicich**
Will now encode the exception strings where user input is used

https://issues.sonatype.org/browse/NEXUS-17816
http://bamboo.s/browse/OSS-DIRF11/latest

### CODE REVIEW

File: directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/RequestException.java [hunk 1]
```diff
@@ -120,12 +121,20 @@ public static RequestException forFormPostMissingParameters(List<String> missing
...
   public static RequestException forSourceNotFound( String sourceName) {
     assert !StringUtils.isEmpty(sourceName);
     
-    return new RequestException( "Unable to find source for '" + sourceName + "'");
+    return new RequestException( "Unable to find source for '" + htmlEncode(sourceName) + "'");
+  }
+
+  private static String htmlEncode(String value) {
+    if (value == null) {
+      return value;
```
Comment (sonatype-zion) [inline directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/RequestException.java]: CRITICAL com.softwarementors.extjs.djn.router.processor.RequestException.htmlEncode(String) may return null, but is declared @Nonnull [rule](http://sonar.s/coding_rules#rule_key=findbugs%3ANP_NONNULL_RETURN_VIOLATION)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RequestException.java#L45](https://github.com/sonatype/directjngine/blob/25c24f6bfbba4c9ca0452211e0f073e6e40d9b2d/directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/RequestException.java#L45): Remove this unused private "RequestException" constructor. [rule](http://sonar.s/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: directjngine/src/main/java/com/softwarementors/extjs/djn/EncodingUtils.java [hunk 2]
```diff
@@ -25,10 +25,20 @@
...
 public final class EncodingUtils {
   private EncodingUtils() {
     // Avoid instantiatin
   }
   
   public static final String UTF8 = "UTF-8";
+
+  public static String htmlEncode(String value) {
+    if (value == null) {
+      return value;
```
[the remark this answers is not in the document]
Comment (wwannemacher) [reply directjngine/src/main/java/com/softwarementors/extjs/djn/EncodingUtils.java]: @dbradicich - this looks different and is my fault... I launched the jenkins build. Jenkins uses a different account to post these comments than bamboo used to use.

Comment (sonatype-zion) [inline directjngine/src/main/java/com/softwarementors/extjs/djn/EncodingUtils.java]: CRITICAL com.softwarementors.extjs.djn.EncodingUtils.htmlEncode(String) may return null, but is declared @Nonnull [rule](http://sonar.s/coding_rules#rule_key=findbugs%3ANP_NONNULL_RETURN_VIOLATION)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RequestException.java#L46](https://github.com/sonatype/directjngine/blob/4b31e04d0b9f1864d5efec65dc7e237485af6bfd/directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/RequestException.java#L46): Remove this unused private "RequestException" constructor. [rule](http://sonar.s/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RequestException.java#L46](https://github.com/sonatype/directjngine/blob/7bcb52a50aa1635e80b464d08bb0acf87bb03746/directjngine/src/main/java/com/softwarementors/extjs/djn/router/processor/RequestException.java#L46): Remove this unused private "RequestException" constructor. [rule](http://sonar.s/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java [hunk 3]
```diff
@@ -18,19 +18,32 @@
...
 
-public class JavaTestsWorkingTest {
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class StandardErrorResponseDataTest
+{
+  @Test
+  public void validateActionEncoded() {
+    //  public StandardErrorResponseData( Long tid, String action, String method, Throwable exception, boolean debugOn ) {
```
Comment (dawidsawa) [inline directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java]: Is this comment really needed?

File: directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java [hunk 4]
```diff
@@ -18,19 +18,32 @@
...
+        "<i onclick=\"alert('haha')\">", new RuntimeException("test"), true);
+    assertThat(data.getAction(), is("&lt;i onclick=&quot;alert('haha')&quot;&gt;"));
+  }
+
   @Test
-  public void test_javaTestsWorking() {
-    // Do nothing: just to see the green bar to make sure our testing system
-    // is well configured!
+  public void validateMethodEncoded() {
+    //  public StandardErrorResponseData( Long tid, String action, String method, Throwable exception, boolean debugOn ) {
```
Comment (dawidsawa) [inline directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java]: Is this comment really needed?

File: directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java (hunk 3 above)
Comment (dbradicich) [reply directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java]: heh, tis not :)

Comment (dbradicich) [reply directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java]: https://github.com/sonatype/directjngine/pull/10/commits/8f08a20132903acccb0347f62861a525174467a8

File: directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java (hunk 4 above)
Comment (dbradicich) [reply directjngine/src/test/java/com/softwarementors/extjs/djn/router/processor/standard/StandardErrorResponseDataTest.java]: https://github.com/sonatype/directjngine/pull/10/commits/8f08a20132903acccb0347f62861a525174467a8

**Comment by dbradicich** [comment]
Changes irrelevant

**Pull request closed without merging**